okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -637,17 +581,14 @@ public void readResponse() throws IOException {
 
       httpStream.writeRequestHeaders(request);
 
-      //Update the networkRequest with the possibly updated interceptor request.
-      networkRequest = request;
```

Comment:
This is slightly a bugfix. If Previously we’d pin the networkRequest in the deepest call to `proceed()`. Now we look at the interceptor’s returned `Response` and take it’s originating request.